Issue 302 Dynamic form training

alexisdgvstlouis opened the pull request:
Inform the ticket you are solving in this pull request: #302

---------------------------------------------------------------

Comment: The part for the 3 new models (question, form and option) are created. 

---------------------------------------------------------------

To have your pull request accepted you have to make sure that:

- [ ] it contains only **well formatted** code, using **4 spaces** for identation and a maximum of **120 characters** per line.
- [ ] classes, methods or function names are **not abreviated**.
- [ ] the code seems **readable for other developers** who will maintain this code in the future.
- [ ] in case complexity is needed, then the code is very **well documented**.
- [ ] new files contain the **license header**.

User Interface:

- [ ] Every section added is illustrated by an icon and described by a text.
- [ ] Every button added contains icon + hint or icon + text.

htmfilho commented:
Numéro excessive des fichiers de migration (5). Il faut réduire à un seul.

alexisdgvstlouis commented:
Bonjour comment j'enlève de github car je ne vois pas les fichiers (comme .DS_Store) en local ? Merci

alexisdgvstlouis commented:
Voilà j'ai enlever les fichiers du github mais il indique ceci quand je fais la dernière commande "git push origin dev" : 
(osis_py3) MacbookAir-Degraeve:osis degraeve$ git push origin dev
To git@github.com:uclouvain/osis.git
 ! [rejected]        dev -> dev (non-fast-forward)

basmot commented:
@alexisdgvstlouis 
Tu ne peux jamais pusher sur la branche Dev, tu dois toujours pusher sur la branche que tu as créée à partir de dev ; git push origin issue#302.

basmot commented:
@alexisdgvstlouis 
Sur ton poste, en local, le plus simple : 
- tu supprimes les fichiers de migration que tu as créé (il y en a 5 en tout il me semble)
- tu supprimes ta DB (dropdb osis_backend_dev)
- tu recrées ta DB (createdb osis_backend_dev)
- Met ton projet Osis à jour avec la branche dev (sur ta branche : git pull origin dev) pour être sûr d'avoir les derniers fichiers de migration (le dernier en dev s'appelle '0026_structure_type.py')
- tu génères le fichier de migration Django (python manage.py makemigrations base). Le nom du fichier généré devrait commencer par '0027_', et il devrait avoir dans son contenu l'attribut dependencies = [
        ('base', '0026_structure_type'),
    ]
- tu effectues les migrations dans ta nouvelle DB vierge (./manage.py migrate)

Si tout s'est bien passé : 
- à la racine du projet, tu fais un 

     git add [nom de tes 5 fichiers de migration supprimés séparés par des espaces + ton nouveau fichier de migration] --all 

(cette commande spécifie à git que tu as supprimé ces 5 fichiers, et que tu en as ajouté un nouveau)
- tu commit tes changements en local (git commit -m "#302 [ton commentaire]")
- tu push tes changements sur le serveur (git push origin issue#302)

alexisdgvstlouis commented:
Merci @basmot  je fais cela en fait j'ai recopié la commande que j'avais vu précédemment.
Je savais pas que cela concernait juste une issue.

basmot commented:
@alexisdgvstlouis Aucun problème.
Le principe de créer une branche est justement de te séparer de la branche principale pour ne pas interférer dessus. 
Quand tu es sur la branche dev, et que tu crées une branche issue#302 (git checkout -b issue#302), Git crée une branche nommée 'issue#302' à partir de la branche 'dev'. L'idée c'est d'effectuer tous tes changements/modifications (commit) sur cette branche issue#302. Cela permet de ne pas toucher à la branche dev pendant ton développement (si qqch foire sur issue#302, cela n'affectera pas dev). Et une fois que tout est testé et fonctionnel sur issue#302, alors on fusionne issue#302 dans dev.

alexisdgvstlouis commented:
J'ai suivi la procédure et je regarde dans Files changed je vois toujours les fichiers supprimé est-ce que j'ai manqué quelque chose?
En faisant git status j'ai bien ceci:
	deleted:    .DS_Store
	deleted:    base/migrations/0024_offerform_offertraining.py
	deleted:    base/migrations/0025_auto_20160329_1456.py
	deleted:    base/migrations/0027_auto_20160329_1527.py
	deleted:    base/migrations/0028_auto_20160329_1544.py
	deleted:    base/migrations/0029_auto_20160329_1544.py

basmot commented:
Vu que tu les as supprimés du répertoire, ces fichiers ne devraient plus être visibles dans ton explorateur de documents. Si ces fichiers sont affichés en vert dans git status, c'est qu'ils sont prêts à être commités. S'ils sont affichés en rouge, c'est que tu dois encore spécifier à Git que tu les as supprimés (pour que Git puisse commiter la suppression).

https://git-scm.com/docs/gittutorial
https://git-scm.com/docs/user-manual.html

alexisdgvstlouis commented:
Les statut sont bien en vert et j'ai comités.

J'ai ceci maintenant :

(osis_py3) MacbookAir-Degraeve:osis degraeve$ git push origin issue#302
Everything up-to-date
(osis_py3) MacbookAir-Degraeve:osis degraeve$ git status
On branch dev
Your branch is ahead of 'origin/dev' by 3 commits.
  (use "git push" to publish your local commits)
nothing to commit, working directory clean

basmot commented:
Tu n'es pas sur la bonne branche ; tu es sur dev là. Je pense que tu as commité sur la mauvaise branche... 
On va continuer ça par mail, car cette zone de commentaire n'est pas un chatter. A chaque fois que l'on commente, les followers de Osis reçoivent un mail de notification... 
Je vais t'envoyer un mail avec les instructions à suivre et les liens vers la doc de Git.
